assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -144,16 +148,21 @@ class PageEdit extends Component {
 
 		if ( VIDEO_BACKGROUND_TYPE === mediaType ) {
 			if ( prevProps.attributes.mediaUrl !== mediaUrl && this.videoPlayer.current ) {
-				if ( this.videoPlayer.current ) {
-					this.videoPlayer.current.load();
-				}
+				this.videoPlayer.current.load();
 			}
 
 			if ( ! poster && ( prevProps.attributes.mediaUrl !== mediaUrl || prevProps.media !== media ) ) {
 				if ( videoFeaturedImage ) {
 					setAttributes( { poster: videoFeaturedImage.source_url } );
 				} else if ( ! mediaId || ( media && ! media.featured_media ) ) {
-					uploadVideoFrame( mediaUrl );
+					/*
```

Comment:
I wonder if we should generate the poster image for already existing videos as well, not only when the media gets updated? E.g. if there's a saved video without the poster image and the Story gets loaded, should we also set the poster image? Perhaps it's OK not to 🤷‍♀ Thoughts?

Reply:
That should actually work already, because `componentDidUpdate` gets called after initialization too.

You can try by checking out `develop`, setting a video, then checking out this branch, building, and reloading the editor.

Reply:
Hmm, maybe, but is shoudln't satisfy this condition :`prevProps.attributes.mediaUrl !== mediaUrl || prevProps.media !== media`?

Just tried with background media and it doesn't seem to generate the poster image.

Reply:
Refactored the code a bit now and removed the logic for "upgrading" existing videos.